packages/services/api/src/modules/audit-logs/providers/audit-logs-types.ts

```diff
@@ -0,0 +1,195 @@
+  targetId: z.string(),
+  schemaName: z.string().nullable().optional(),
+  schemaSdl: z.string(),
+});
+
+const SchemaDeletedAuditLogSchema = z.object({
+  projectId: z.string(),
+  targetId: z.string(),
+  schemaName: z.string(),
+});
+
+const RoleCreatedAuditLogSchema = z.object({
+  roleId: z.string(),
+  roleName: z.string(),
+});
+
+const RoleAssignedAuditLogSchema = z.object({
+  roleId: z.string(),
+  roleName: z.string(),
+  userIdAssigned: z.string(),
+  userEmailAssigned: z.string(),
+});
+
+const RoleDeletedAuditLogSchema = z.object({
+  roleId: z.string(),
+  roleName: z.string(),
+});
+
+const RoleUpdatedAuditLogSchema = z.object({
+  roleId: z.string(),
+  roleName: z.string(),
+  updatedFields: z.string(),
+});
+
+export const auditLogSchema = z.discriminatedUnion('eventType', [
+  z.object({
+    eventType: z.literal('USER_INVITED'),
+    UserInvitedAuditLogSchema,
+  }),
+  z.object({
+    eventType: z.literal('USER_JOINED'),
+    UserJoinedAuditLogSchema,
+  }),
+  z.object({
+    eventType: z.literal('USER_REMOVED'),
+    UserRemovedAuditLogSchema,
+  }),
+  z.object({
+    eventType: z.literal('ORGANIZATION_SETTINGS_UPDATED'),
+    OrganizationSettingsUpdatedAuditLogSchema,
+  }),
+  z.object({
+    eventType: z.literal('ORGANIZATION_TRANSFERRED'),
+    OrganizationTransferredAuditLogSchema,
+  }),
+  z.object({
+    eventType: z.literal('PROJECT_CREATED'),
+    ProjectCreatedAuditLogSchema,
+  }),
+  z.object({
+    eventType: z.literal('PROJECT_SETTINGS_UPDATED'),
+    ProjectSettingsUpdatedAuditLogSchema,
+  }),
+  z.object({
+    eventType: z.literal('PROJECT_DELETED'),
+    ProjectDeletedAuditLogSchema,
+  }),
+  z.object({
+    eventType: z.literal('TARGET_CREATED'),
+    TargetCreatedAuditLogSchema,
+  }),
+  z.object({
+    eventType: z.literal('TARGET_SETTINGS_UPDATED'),
+    TargetSettingsUpdatedAuditLogSchema,
+  }),
+  z.object({
+    eventType: z.literal('TARGET_DELETED'),
+    TargetDeletedAuditLogSchema,
+  }),
+  z.object({
+    eventType: z.literal('SCHEMA_POLICY_SETTINGS_UPDATED'),
+    SchemaPolicySettingsUpdatedAuditLogSchema,
+  }),
+  z.object({
+    eventType: z.literal('SCHEMA_CHECKED'),
+    SchemaCheckedAuditLogSchema,
+  }),
+  z.object({
+    eventType: z.literal('SCHEMA_PUBLISH'),
+    SchemaPublishAuditLogSchema,
+  }),
+  z.object({
+    eventType: z.literal('SCHEMA_DELETED'),
+    SchemaDeletedAuditLogSchema,
+  }),
+  z.object({
+    eventType: z.literal('ROLE_CREATED'),
+    RoleCreatedAuditLogSchema,
+  }),
+  z.object({
+    eventType: z.literal('ROLE_ASSIGNED'),
+    RoleAssignedAuditLogSchema,
+  }),
+  z.object({
+    eventType: z.literal('ROLE_DELETED'),
+    RoleDeletedAuditLogSchema,
+  }),
+  z.object({
+    eventType: z.literal('ROLE_UPDATED'),
+    RoleUpdatedAuditLogSchema,
+  }),
+]);
+
+export type AuditLogEvent = z.infer<typeof auditLogSchema> & {
+  id?: string;
+  eventTime: string;
+  user: {
+    userId: string;
+    userEmail: string;
+    user?: User | null;
```

Comment:
prefer explicit null 
```suggestion
    user: User | null;
```

Reply:
Done! thanks